qCC/ccPrimitiveDistanceDlg.cpp

```diff
@@ -0,0 +1,64 @@
+//##########################################################################
+//#                                                                        #
+//#                              CLOUDCOMPARE                              #
+//#                                                                        #
+//#  This program is free software; you can redistribute it and/or modify  #
+//#  it under the terms of the GNU General Public License as published by  #
+//#  the Free Software Foundation; version 2 or later of the License.      #
+//#                                                                        #
+//#  This program is distributed in the hope that it will be useful,       #
+//#  but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          #
+//#  GNU General Public License for more details.                          #
+//#                                                                        #
+//#          COPYRIGHT: EDF R&D / TELECOM ParisTech (ENST-TSI)             #
+//#                                                                        #
+//##########################################################################
+
+#include "ccPrimitiveDistanceDlg.h"
+
+//Qt
+#include <QHeaderView>
+#include <QMessageBox>
+
+
+//System
+#include <assert.h>
+
+ccPrimitiveDistanceDlg::ccPrimitiveDistanceDlg(QWidget* parent)
+	: QDialog(parent, Qt::Tool)
+	, Ui::primitiveDistanceDlg()
+{
+	setupUi(this);
+
+	
+
+	signedDistCheckBox->setChecked(true);
+	flipNormalsCheckBox->setEnabled(true);
+	
+	connect(cancelButton, SIGNAL(clicked()), this, SLOT(cancelAndExit()));
+	connect(okButton, SIGNAL(clicked()), this, SLOT(applyAndExit()));
+	connect(signedDistCheckBox, SIGNAL(toggled(bool)), this, SLOT(toggleSigned(bool)));
+	show();
```

Comment:
I had started from the ccComparisonDlg as an example the constructor there contained show and processEvents assuming because of the computeApproxDistances call afterwards

Reply:
Then we have to fix it as well :D

Reply:
These should also be using Qt5 style connects, not the old SIGNAL and SLOT.

Reply:
I'm happy to update them, but I'm missing something in my understanding, Ive added them with actions before but not successful for Qpushbutton 

``` C++
connect(cancelButton, &QPushButton::clicked, this, ccPrimitiveDistanceDlg::cancelAndExit());
```
I am missing an overload.

@asmaloney  have an example I can follow?

Reply:
never mind I think I got it I had the extra () at the end of my slot function